File: baekjoon/15649_N과 M (1)/B_15649.java
package baekjoon_java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class B_15649 {
	
	private static int N;
	
	private static int M;
	
	private static int[] check;
	
	private static int[] arr;
	
	private static void permutation(int idx) {
		if (idx == M) {
			for (int i = 0; i < M; i++) {
				System.out.print(arr[i] + " ");
			}
			System.out.println();
		} else {
			for (int i = 0; i < N; i++) {
				if (check[i] == 0) {
					check[i] = 1;
					arr[idx] = i + 1;
					permutation(idx + 1);
					check[i] = 0;
				}
			}
		}
	}
	
	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		N = Integer.parseInt(st.nextToken());
		M = Integer.parseInt(st.nextToken());
		check = new int[N];
		arr = new int[N];
		permutation(0);
	}

}

File: baekjoon/15649_N과 M (1)/B_15649.js
const permutation = function(idx, m) {
    if (idx === m) {
        console.log(perm.join(" "))
    } else {
        for (let j = 0; j < N; j++) {
            if (check[j] === 0) {
                perm.push(arr[j])
                check[j] = 1
                permutation(idx + 1, m)
                perm.splice(perm.length - 1, 1)
                check[j] = 0
            }
        }
    }
}

const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim()

const nAndM = input.split(' ')
const N = parseInt(nAndM[0])
const M = parseInt(nAndM[1])

const arr = new Array(N)
for (let i = 1; i <= N; i++) {
    arr[i - 1] = i
}

let check = new Array(N).fill(0)
let perm = []

permutation(0, M)